Document use of Windows Credential Store from WSL

kwessel opened the pull request.

edthedev commented on docs/readme/body.rst:
```diff
@@ -554,6 +557,46 @@ Environment Variables
+to the Windows Python executable. If you need to find the full path
+to this executable, run this from the Windows command prompt:
+
+    where python.exe
```
```suggestion
    which python.exe
```

When I was stepping through, I had to swap this for `which`.

kwessel replied:
> When I was stepping through, I had to swap this for `which`.

That's because you ran it inside WSL. The instructions say to run that from a Windows command prompt. Which is not a Windows command, but where is. If you do have which on Windows, I'd like to know where you got it.

edthedev replied:
@kwessel Oh! Since my default shell is `PowerShell`, `where` did not work for me. It works fine in Windows Command Prompt. In `PowerShell`, `Get-Command python.exe` works - but still requires translation.

A nice thing about running `which python.exe` inside WSL, is that it gave me the WSL-friendly path, without any translation. It only works because I included the `.exe`, since `which python` stops at the version inside WSL.

kwessel replied:
which python.exe doesn't work for me inside wsl. It can't find python.exe, presumably because I don't have any /mnt/c directories in my search path. I imagine most people won't. I went with the where command since that was a lowest common denominator. One could also do /find /mnt/c -name python.exe inside wsl, but that takes a very long time to run and produces a lot of noise. So, I went with the Windows command prompt and the "where" command.

edthedev replied:
That's a great point. I did have all of `/mnt` on my path, but I do typically remove it. 

@ddriddle `which python.exe` worked for me because I had not removed the weird `/mnt` path stuff from my profile yet. 

The approach @kwessel included will be more reliable. I made a separate suggestion for emphasis to help others not miss that the command is for Windows Command Prompt.

edthedev commented on docs/readme/body.rst:
```diff
@@ -554,6 +557,46 @@ Environment Variables
+Inside WSL, set this environment variable like this. If Python is at
+c:\path\to\python.exe, translate the path like this:
+
+    export KEYRING_PROPERTY_PYTHON=/mnt/c/paty/to/python.exe
```
```suggestion
    export KEYRING_PROPERTY_PYTHON='/mnt/c/paty/to/python.exe'
```

My path contained a space, which will be common for certain Windows versions.

kwessel replied:
> My path contained a space, which will be common for certain Windows versions.

Good catch, thank you. I've implemented that change and also corrected the typo of /mnt/c/paty to /mnt/c/path.

edthedev commented on docs/readme/body.rst:
```diff
@@ -554,6 +557,46 @@ Environment Variables
+
+Finally, set the KEYRING_PROPERTY_PYTHON envorinment variableto point
+to the Windows Python executable. If you need to find the full path
+to this executable, run this from the Windows command prompt:
```
```suggestion
to this executable, run this from the **Windows command prompt**:
```

Suggest emphasis here, since I missed it and ran the command in PowerShell.

ddriddle commented on docs/readme/body.rst:
```diff
@@ -444,7 +444,10 @@ enable_keyring
     Locker, and Linux keyrings. Additional, system configuration
     may be required to use a keyring on Linux systems (See
-    https://pypi.org/project/keyring for details). Set to True to
+    https://pypi.org/project/keyring for details). For an example ofthis
+    WSL and the Windows Credential Store,see below.
```
```suggestion
    https://pypi.org/project/keyring for details). For an example of this
    on WSL using the Windows Credential Store, see
    `Keyrings with WSL and Windows Credential Store`_ below.
```

ddriddle commented on docs/readme/body.rst:
```diff
@@ -554,6 +557,46 @@ Environment Variables
+
+    pip install keyring
+
+Next, inside WSL, install the keyring-pybridge module:
```
```suggestion
Next, inside WSL, install the keyring-pybridge module::
```

ddriddle commented on docs/readme/body.rst:
```diff
@@ -554,6 +557,46 @@ Environment Variables
+
+Still in WSL, Set the PYTHON_KEYRING_BACKEND environment variable to
+tell Python to use the pybridge keyring backend. This is shell
+specific but might look like:
```
```suggestion
specific but might look like::
```

ddriddle reviewed (changes requested):
@kwessel thank you very much for the PR! I have added suggestion to correct some typos, mostly missing white space and a couple spelling errors. I also added suggestions to add Unix and Windows prompts for the various examples. I have prompts throughout the document elsewhere so for consistency they need to be here as well. I think it also helps with readability since there is switching back and forth between WSL and Windows CMD.EXE.

kwessel commented:
> @kwessel thank you very much for the PR! I have added suggestion to correct some typos, mostly missing white space and a couple spelling errors. I also added suggestions to add Unix and Windows prompts for the various examples. I have prompts throughout the document elsewhere so for consistency they need to be here as well. I think it also helps with readability since there is switching back and forth between WSL and Windows CMD.EXE.

So sorry about the typos, thanks for cleaningthose up! And yes, completely agreed on adding prompts. All changes have been applied. Feel free to squash and merge whenever you like, but I'll leave that up to you.

ddriddle reviewed (changes requested):
Everything looks great except for a formatting error. Path's need escaping to show up correctly in RST format.

ddriddle commented on docs/readme/body.rst:
```diff
@@ -554,6 +558,46 @@ Environment Variables
     look for configuration files in (``/tmp/.aws-login/``).
 
+Keyrings with WSL and Windows Credential Store
+=====================================
```
This needs to be the same length as 561. It is considered an error on PyPI:

```
twine check --strict dist/awscli_login-1.1.dev22-py3-none-any.whl dist/awscli_login-1.1.dev22.tar.gz
Checking dist/awscli_login-1.1.dev22-py3-none-any.whl: FAILED
ERROR    `long_description` has syntax errors in markup and would not be        
         rendered on PyPI.                                                      
         line 640: Warning: Title underline too short.                          
                                                                                
         Keyrings with WSL and Windows Credential Store                         
         =====================================
```

This should fix the issue:

```suggestion
==============================================
```

ddriddle commented on docs/readme/body.rst:
```diff
@@ -554,6 +558,46 @@ Environment Variables
+==============================================
+
+If running under WSL, you may wish to store credentials in the Windows
+Credential Store in the Windoes host operating system and access them
```
```suggestion
Credential Store in the Windows host operating system and access them
```

ddriddle commented on docs/readme/body.rst:
```diff
@@ -554,6 +558,46 @@ Environment Variables
+Credential Store in the Windoes host operating system and access them
+from within WSL. To do this, first install Python on the Windows
+host operating system if not already installed. You can do this from
+the Microsoft Store or using Winget, or you can download the installer
```
```suggestion
the Microsoft Store or using WinGet, or you can download the installer
```

ddriddle reviewed (approved):
Looks great. A few typos need fixing. I will fix then merge.
